test/EFCore.PG.FunctionalTests/Scaffolding/NpgsqlDatabaseModelFactoryTest.cs

```diff
@@ -1374,12 +1378,14 @@ public void Warn_missing_principal_table_for_foreign_key()
                     Assert.Empty(dbModel.Sequences);
 
                     // Now make sure the field itself is properly reverse-engineered.
-                    var column = dbModel.Tables.Single().Columns.Single();
-                    Assert.Equal(ValueGenerated.OnAdd, column.ValueGenerated);
-                    Assert.Null(column.DefaultValueSql);
-                    Assert.Equal(NpgsqlValueGenerationStrategy.SerialColumn, (NpgsqlValueGenerationStrategy)column[NpgsqlAnnotationNames.ValueGenerationStrategy]);
+                    foreach (var column in dbModel.Tables.Select(t => t.Columns.Single()))
```

Comment:
Seems like this is throwing. Some unanticipated table being picked up?

Reply:
Hmm, it's passing for me after running the entire class several times. Can you try dropping the database and retrying? Maybe you have some leftover from the distant past...